infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -54,7 +66,7 @@ public void write(@Nonnull K key, @Nonnull V value) {
         try {
             CompletableFuture<Void> cf = new CompletableFuture();
             operationsQueue.add(new BatchWriterOperation(BatchWriterOperation.Type.WRITE,
-                    (Long) key, (LogData) value, null, cf));
+                    (Long) key, (LogData) value, ((LogData) value).getEpoch(), null, cf));
```

Comment:
Considering the caller is already passing value (LogData), can we avoid passing epoch by client and extract the epoch and set relevant field within BatchWriterOperation? I know that BatchWriterOperation is using lomok's @Data. hence, this entails some modification of the constructor but probably is worth it as it will make it more concise from the caller's perspective.

Reply:
As discussed, this can be done, but `write` being the only use case, it would be more work.